packages/swing-store-lmdb/src/lmdbSwingStore.js

```diff
@@ -149,19 +156,77 @@ function makeSwingStore(dirPath, forceReset = false) {
     }
   }
 
-  const storage = {
+  const kvStore = {
     has,
     getKeys,
     get,
     set,
     delete: del,
   };
 
+  /**
+   * Generator function that returns an iterator over the items in a stream.
+   *
+   * @param {string} streamName  The stream to read
+   *
+   * @yields {string} an iterator for the items in the named stream
+   */
+  function* readStream(streamName) {
+    try {
+      const fd = fs.openSync(`${dirPath}/${streamName}`, 'r');
+      if (fd) {
+        const reader = new Readlines(fd);
+        while (true) {
+          const line = /** @type {string|false} */ (reader.next());
+          if (line) {
+            yield line;
+          } else {
+            break;
+          }
+        }
+        fs.closeSync(fd);
+      }
+    } catch (e) {
+      if (e.code !== 'ENOENT') {
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Append an item to a stream.
+   *
+   * @param {string} streamName  The stream to append to
+   * @param {string} item  The item to append to it
+   */
+  function appendItem(streamName, item) {
+    assert.typeof(
+      streamName,
+      'string',
+      X`non-string stream name ${streamName}`,
+    );
+    assert(streamName.match(/^[-\w]+$/));
```

Comment:
I highly approve of this paranoia. Let's add the same check to the `readStream` side, so this doesn't enable the kernel to read arbitrary parts of the filesystem.

Reply:
Let's also assert that the item is a string, and that there are no newlines inside it, to preserve the format expected by `readStream`.

(we should put these checks into swing-store-simple too)